lib/generate-report.js

```diff
@@ -444,6 +456,33 @@ function generateReport(options) {
             );
         });
     }
+    
+    // From https://gist.github.com/remino/1563878
+    // used to make durations that are more easily read.
+    function convertMS(ms) {
+        var d, h, m, s, ms;
+        ms = ms;
+        s = Math.floor(ms / 1000);
+        ms = ms % 1000;
+        m = Math.floor(s / 60);
+        s = s % 60;
+        h = Math.floor(m / 60);
+        m = m % 60;
+        d = Math.floor(h / 24);
+        h = h % 24;
+        return { d: d, h: h, m: m, s: s, ms: ms};
+    }
+
+    function formatDuration(ms) {
+        const timeData = convertMS(ms);
+        var d, h, m, s, ms;
+        ms = timeData.ms ? timeData.ms + "ms" : "";
+        s = timeData.s ? timeData.s + "s" : "";
+        m = timeData.m ? timeData.m + "m" : "";
+        h = timeData.h ? timeData.h + "h" : "";
+        d = timeData.d ? timeData.d + "d" : "";
+        return d + h + m + s + ms;
```

Comment:
I've been thinking about this and I wonder what the _best_ layout would be, or yours

![image](https://user-images.githubusercontent.com/11979740/36395835-7ebfbe16-15bb-11e8-849f-f35fb85697b4.png)

or this

![image](https://user-images.githubusercontent.com/11979740/36395829-73704f8a-15bb-11e8-9889-26fe0539bf75.png)

Personally I prefer the last, even though there is no d/h/m/s/ms notation in it.

Reply:
This is fine by me, I didn't think of different ways to display and just used the format that the Cucumber Jenkins Plugin uses.